src/fmllauncher/java/net/minecraftforge/fml/loading/progress/ClientVisualization.java

```diff
@@ -223,42 +217,47 @@ private void renderMessage(final String message, final float[] colour, int row,
     }
 
     private void closeWindow() {
+        LOGGER.debug("Destroying early loading screen");
         glfwFreeCallbacks(window);
         glfwDestroyWindow(window);
+        GL.setCapabilities(null);
         glfwTerminate();
         glfwSetErrorCallback(null).free();
     }
 
     @Override
     public void start() {
         thread = new Thread(this::run);
+        thread.setUncaughtExceptionHandler((t, e) -> LOGGER.error("Exception rendering early loading screen!", e));
+        thread.setName("Early loading screen renderer");
         thread.setDaemon(true);
         thread.start();
     }
 
     private void run() {
         initWindow();
-        running = true;
-        while (running) {
-            renderProgress();
-            try {
-                Thread.sleep(50);
-            } catch (InterruptedException e) {
+        try {
+            while (running) {
+                renderProgress();
+                try {
+                    Thread.sleep(50);
```

Comment:
`glfwSwapInterval` already enabled vsync so it adds internal sleep already. The maximum possible wait time should be about 1/6th of a second. This sleep just makes it possible to do an interrupt, but in case the code is already in `glfwSwapBuffers`, this won't do anything.

Does it actually add any significant improvement? I would expect destroying the window to take more of the time.

Reply:
Destroying the window takes about ~30-80ms on my machine, waiting for the other thread to respond to us setting running to false with the interrupt about ~1-10ms (from looking at the newly added logs). The interrupt does nothing major, but it's an easy thing to save some millis